contracts/libraries/InteractionLogic.sol

```diff
@@ -45,7 +45,7 @@ library InteractionLogic {
         mapping(bytes32 => uint256) storage _profileIdByHandleHash
     ) external {
         if (profileIds.length != followModuleDatas.length) revert Errors.ArrayMismatch();
-        for (uint256 i = 0; i < profileIds.length; ++i) {
+        for (uint256 i = 0; i < profileIds.length; ) {
```

Comment:
here you go with the length from the calldata. Better to be consistent and apply the same "mechanism" here I guess

Reply:
Unfortunately we run into a stack too deep error if we do that here :(